tools/code-tools/asmtools.sh

```diff
@@ -1,80 +1,62 @@
-
-    echo "Copying release/lib/asmtools*.jar file to ../"
-    cp release/lib/asmtools*.jar ..
-
-    # WORKSPACE/asmtools/build/BUILD_DIR/dist
-    echo "Moving into $artifact-build/dist"
-    cd dist
-
-    echo "tar-ing ${artifact}.zip into ${artifact}.tar"
-    tar fcv "$artifact.tar" *.zip
-
-    echo "Moving ${artifact}.tar to ../.."
-    mv "$artifact.tar" ../..
-    # WORKSPACE/asmtools/build/
-    cd ../..
-
-    echo "gzipping ${artifact}.tar"
-    gzip -9 -f "${artifact}.tar"
-
-    if [ -d "$artifact-build" ];
-    then
-        mv "$artifact-build" asmtools/
-    fi
-
-    echo "Creating checksums for asmtools.jar and ${artifact}.tar.gz"
-    sha256sum asmtools.jar > asmtools.jar.sha256sum.txt
-    sha256sum "${artifact}.tar.gz" > "${artifact}.tar.gz.sha256sum.txt"
-
-    echo "Moving into asmtools"
-    cd asmtools
+    local  branchOrTag=${1}
+    export JAVA_HOME=${2}
+    git checkout $branchOrTag
+    echo "Moving into maven build..."
+    pushd maven
+      echo "Removing asmtools $branchOrTag old maven wrapper"
+      rm -rf src target pom.xml
+      #export JAVA_TOOL_OPTIONS="-Dfile.encoding=UTF8 -Xdoclint:none -Dmaven.javadoc.skip=true -Dgpg.skip"
+      echo "Generating asmtools $branchOrTag maven wrapper"
+      sh mvngen.sh
+      echo "Cleaning possible previous run"
+      mvn clean
+      echo "Running asmtools $branchOrTag tests by $JAVA_HOME"
+      mvn test || echo "Test now correctly fails, this have to be fixed upstream"
+      echo "Running asmtools $branchOrTag build by $JAVA_HOME"
+      mvn package -DskipTests # mvn install will do much more, but I doubt we wish that (javadoc, sources, gpg sign...)
+      echo "Moving down to target"
+      pushd target
+        echo "Asmtools $branchOrTag artifact:"
+        ls -l
+        local mainArtifact=`ls asmtools*.jar`
+        echo "Copying maven/target/$mainArtifact file to WORKSPACE($WORKSPACE)"
+        cp  $mainArtifact $WORKSPACE
+        if [ -e surefire-reports ] ; then
+          local testResults=`echo $mainArtifact | sed "s/.jar/-tests.tar.gz/"`
+          echo "Compressing and archiving test results as $testResults"
+          tar -czf $testResults surefire-reports
+          echo "Copying maven/target/$testResults file to WORKSPACE($WORKSPACE)"
+          cp  $testResults $WORKSPACE
+        else
+          echo "No test results!"
+        fi
+      popd
+    popd
 }
 
-cd asmtools
-
-export PRODUCT_VERSION=$(grep "PRODUCT_VERSION     \= " build/productinfo.properties | awk '{print $3}')
-# shellcheck disable=SC2005,SC2046
-export BUILD_DIR=$(echo $(eval echo $(grep "BUILD_DIR = " build/build.properties | awk '{print $3}')))
-
-generateArtifact "tip" "${PRODUCT_VERSION}"
+if [ ! -e asmtools ] ; then
+  git clone https://github.com/openjdk/asmtools.git
+fi
+
+jdk08=`find /usr/lib/jvm/ -maxdepth 1 | sort | grep -e -1.8.0-  -e jdk-8   | head -n 1`
+jdk17=`find /usr/lib/jvm/ -maxdepth 1 | sort | grep -e -17-     -e jdk-17  | head -n 1`
```

Comment:
I don't think this work. I'm still getting the JAVA_HOME error for jdk17. Maybe:

`jdk17=`find /usr/lib/jvm/ -maxdepth 1 | sort | grep -e jdk-17- -e jdk-17 | head -n 1`